apps/settings/js/modules/date_time.js

```diff
@@ -197,8 +194,6 @@ define(function(require) {
         break;
       case 'localized':
         var d = new Date();
-        // update time format while language changed
-        this._updateTimeFormat();
```

Comment:
IIUC, this is the core of the pull request?  If the user changes the languages two things can happen:

1. if hour12 was set manually, it persists and overrides the default setting for the new language,
2. if hour12 was set to 'automatic', `navigator.mozHour12` is `undefined` and `this._formatTime` will use Intl's settings for the new language.

Is this correct?